system/Database/Forge.php

```diff
@@ -482,7 +482,7 @@ public function createTable($table, $if_not_exists = false, array $attributes =
 
 		if (($result = $this->db->query($sql)) !== false)
 		{
-			empty($this->db->dataCache['table_names']) || $this->db->dataCache['table_names'][] = $table;
+			empty($this->db->dataCache['table_names']) OR $this->db->dataCache['table_names'][] = $table;
```

Comment:
Was this change actually required? Typically we keep to the bitwise || unless needed.

Reply:
If I keep it || then editor gives an error "Cannot use [] for reading". So changed

Reply:
I went through the code flow. The flow should be like if table_names is empty then assign table name to it.
But I run the dummy code,
$a = [];
empty($a) OR $a[] = 5;
print_r($a);

$a = [];
empty($a) || $a[] = 5;
print_r($a);

Both are giving me same output as blank array.

But the code should be,
if(empty($this->db->dataCache['table_names']))
{
        $this->db->dataCache['table_names'][] = $table;
}

Correct me If I am wrong.

Reply:
I think you're correct here. This is code ported from CI3 that I hadn't really touched much before. Not a fan of the original coding style there, honestly, but your fix looks correct.

Reply:
Should I generate a PR for correction of this condition like I stated in earlier comment

Reply:
Yes, that would be great. Thanks.